### ISSUE TITLE
Boot issues on Debian 12 based SONiC images on a Wedge100-32X

### ISSUE THREAD

**Issue by d3dx9** [sonic-net/sonic-buildimage#20475]
#### Description

Kernel panic on boot due to i915 driver on Wedge100-32X

#### Steps to reproduce the issue:
1. Install SONiC based on Debian 12
2. Boot

#### Describe the results you received:
See attached file

#### Describe the results you expected:
A booting image, it works as soon I add module_blacklist=i915 to the boot parameters in grub.

#### Output of `show version`:

admin@sonic:~$ show version

SONiC Software Version: SONiC.202405.663939-9e6879fc3
SONiC OS Version: 12
Distribution: Debian 12.6
Kernel: 6.1.0-22-2-amd64
Build commit: 9e6879fc3
Build date: Wed Oct  9 13:02:43 UTC 2024
Built by: azureuser@bbfadc86c00000A

Platform: x86_64-facebook_wedge100-r0
HwSKU: Facebook-W100-C32
ASIC: broadcom
ASIC Count: 1
Serial Number: N/A
Model Number: N/A
Hardware Revision: N/A
Uptime: 09:54:42 up 10 min,  2 users,  load average: 1.46, 1.84, 1.31
Date: Sun 16 Jun 2024 09:54:42

Docker images:
REPOSITORY                    TAG                       IMAGE ID       SIZE
docker-syncd-brcm             202405.663939-9e6879fc3   489317568b5a   717MB
docker-syncd-brcm             latest                    489317568b5a   717MB
docker-gbsyncd-broncos        202405.663939-9e6879fc3   78ca8f2e7e6c   354MB
docker-gbsyncd-broncos        latest                    78ca8f2e7e6c   354MB
docker-gbsyncd-credo          202405.663939-9e6879fc3   56e8538079bb   327MB
docker-gbsyncd-credo          latest                    56e8538079bb   327MB
docker-orchagent              202405.663939-9e6879fc3   70639374fcfb   357MB
docker-orchagent              latest                    70639374fcfb   357MB
docker-sflow                  202405.663939-9e6879fc3   96fe999d3479   345MB
docker-sflow                  latest                    96fe999d3479   345MB
docker-nat                    202405.663939-9e6879fc3   de6e275b5bf7   347MB
docker-nat                    latest                    de6e275b5bf7   347MB
docker-fpm-frr                202405.663939-9e6879fc3   685f085fd52d   376MB
docker-fpm-frr                latest                    685f085fd52d   376MB
docker-dhcp-relay             latest                    b2f7652f5bbf   325MB
docker-macsec                 latest                    6c3550d71270   347MB
docker-snmp                   202405.663939-9e6879fc3   f6c9cf1e0d34   355MB
docker-snmp                   latest                    f6c9cf1e0d34   355MB
docker-platform-monitor       202405.663939-9e6879fc3   6d7e67799672   442MB
docker-platform-monitor       latest                    6d7e67799672   442MB
docker-teamd                  202405.663939-9e6879fc3   9c57c2d09dde   344MB
docker-teamd                  latest                    9c57c2d09dde   344MB
docker-router-advertiser      202405.663939-9e6879fc3   4e115c55bd0d   316MB
docker-router-advertiser      latest                    4e115c55bd0d   316MB
docker-mux                    202405.663939-9e6879fc3   a2580af25222   368MB
docker-mux                    latest                    a2580af25222   368MB
docker-lldp                   202405.663939-9e6879fc3   3cc4799ebe66   361MB
docker-lldp                   latest                    3cc4799ebe66   361MB
docker-sonic-gnmi             202405.663939-9e6879fc3   01720ac74e88   400MB
docker-sonic-gnmi             latest                    01720ac74e88   400MB
docker-sonic-mgmt-framework   202405.663939-9e6879fc3   b59d40b78b5f   402MB
docker-sonic-mgmt-framework   latest                    b59d40b78b5f   402MB
docker-database               202405.663939-9e6879fc3   d18f7550c887   324MB
docker-database               latest                    d18f7550c887   324MB
docker-eventd                 202405.663939-9e6879fc3   cc6496bd72a2   316MB
docker-eventd                 latest                    cc6496bd72a2   316MB
[message.txt](https://github.com/user-attachments/files/17342382/message.txt)

**Comment by vdahiya12** [comment]
@d3dx9 please help as we don't maintain wedge platforms.

**Comment by leeminhow** [comment]
> [@d3dx9](https://github.com/d3dx9) please help as we don't maintain wedge platforms.

This issue is most likely a firmware issue. I am running the same wedge100.

Before updating the firmware (Facebook's default firmware), the i915 kernel panic occurred like d3dx9, and it disappeared after applying the blacklist.

After updating the BIOS firmware, the i915 kernel panic did not occur.

However, there is an issue where the device name is changed from facebook wedge100 to accton wedge100. This causes an issue where the HwSKU cannot be loaded.

In fact, facebook wedge100 and accton wedge100 seem to have exactly the same hardware.

**Comment by leeminhow** [comment]
@vdahiya12 I have additional questions.

I have a Wedge100 in my home lab and SONiC is an alternative to make it work.

However, for Wedge100 in SONiC, the sonic-platform is not implemented at all, so I can't do almost anything like read status of power supply, read transceiver's eeprom, etc......

How can you help me implement the sonic platform for wedge100?

Thank you.

**Comment by d3dx9** [comment]
@leeminhow - would you mind to share the firmware / bios? I tried with the EdgeCore one but sadly this bricked the whole board, better said ONIE has problems to be installed because it does not detect the SSD correctly anymore.

**Comment by leeminhow** [comment]
@d3dx9 Did you apply the firmware from the link below?
https://support.edge-core.com/hc/en-us/articles/360034506274-Wedge100-32X-BIOS-R1-00-E3-latest

some people get confused between "Wedge100BF-32X" and "Wedge100-32X" and upload the wrong firmware.

ONIE is also provided by edgecore, but the HwSKU is edgecore, not facebook.
https://support.edge-core.com/hc/en-us/articles/360034506594-Wedge100-32X-ONIE-v2017-02-00-06-latest


I'm on vacation so I think I'll be able to write up how to update the firmware and change the HwSKU to install SONiC.
I'll write it up within 3 days and attach the link.



-The translation may be a bit strange. Thank You!

**Comment by d3dx9** [comment]
Yes, I did upload the correct update. And it also seems to boot, but as soon I try to install ONIE it also installs it, but afterwards it does not boot and hang after the 3 lines with the model.
